src/pip/_internal/commands/debug.py

```diff
@@ -224,7 +247,11 @@ def run(self, options, args):
         show_value("pip._vendor.certifi.where()", where())
         show_value("pip._vendor.DEBUNDLED", pip._vendor.DEBUNDLED)
 
-        show_vendor_versions()
+        if not pip._vendor.DEBUNDLED:
+            show_vendor_versions()
```

Comment:
The whole point of `show_vendor_versions()` is to be called in the `pip._vendor.DEBUNDLED` case (the `else` branch here).
If pip vendoring is not debundled, we only need to check `vendor.txt` content.